src/services/ansiblePlaybook.ts

```diff
@@ -0,0 +1,179 @@
+          'ansible syntax-check',
+          undefined,
+          'Processing files...'
+        );
+      }
+
+      const [command, env] = withInterpreter(
+        `${settings.ansible.path}-playbook`,
+        `${docPath} --syntax-check`,
+        settings.python.interpreterPath,
+        settings.python.activationScript
+      );
+
+      await exec(command, {
+        encoding: 'utf-8',
+        cwd: workingDirectory,
+        env: env,
+      });
+    } catch (error) {
+      if (error instanceof Error) {
+        const execError = error as ExecException & {
+          // according to the docs, these are always available
+          stdout: string;
+          stderr: string;
+        };
+
+        // This is the regex to extract the filename, line and column number from the strerr produced by syntax-check command
+        const ansibleSyntaxCheckRegex =
+          /The error appears to be in '(?<filename>.*)': line (?<line>\d+), column (?<column>\d+)/;
+
+        const result = ansibleSyntaxCheckRegex.exec(execError.stderr);
+
+        diagnostics = result
+          ? this.processReport(
+              execError.message,
+              result.groups.filename,
+              parseInt(result.groups.line),
+              parseInt(result.groups.column)
+            )
+          : this.processReport(execError.message, docPath, 1, 1);
+
+        if (execError.stderr) {
+          this.connection.console.info(
+            `[ansible syntax-check] ${execError.stderr}`
+          );
+        }
+      } else {
+        this.connection.console.error(
+          `Exception in AnsibleSyntaxCheck service: ${JSON.stringify(error)}`
+        );
+      }
+    }
+
+    if (progressTracker) {
+      progressTracker.done();
+    }
+    return diagnostics;
+  }
+
+  private processReport(
+    result: string,
+    fileName: string,
+    line: number,
+    column: number
+  ): Map<string, Diagnostic[]> {
+    const diagnostics: Map<string, Diagnostic[]> = new Map();
+    if (!result) {
+      this.connection.console.warn(
+        'Standard output from ansible syntax-check is suspiciously empty.'
+      );
+      return diagnostics;
+    }
+    try {
+      const start: Position = {
+        line: line - 1,
+        character: column - 1,
+      };
+      const end: Position = {
+        line: line - 1,
+        character: Number.MAX_SAFE_INTEGER,
+      };
+      const range: Range = {
+        start: start,
+        end: end,
+      };
+
+      const severity: DiagnosticSeverity = DiagnosticSeverity.Error;
+
+      const locationUri = `file://${fileName}`;
+
+      let fileDiagnostics = diagnostics.get(locationUri);
+      if (!fileDiagnostics) {
+        fileDiagnostics = [];
+        diagnostics.set(locationUri, fileDiagnostics);
+      }
+
+      const message: string = result;
+      fileDiagnostics.push({
+        message: message,
+        range: range || Range.create(0, 0, 0, 0),
+        severity: severity,
+        source: 'Ansible',
+      });
```

Comment:
I'm failing to see how anything in this block could raise an error. Could you explain why you expect any here?

Reply:
Looks like a copy-paste leftover - there is no parsing happening here

Reply:
Yep, Priyam and I discussed this on a 1:1 call yesterday and I think we tried to address it then in a pair programming session. @priyamsahoo — did you push all the local commits to this PR? It'd be useful to do this so that *everybody* sees that there's no point in discussing the outdated code anymore.